core/src/main/java/c/org/rajawali3d/control/RenderControl.java

```diff
@@ -0,0 +1,212 @@
+package c.org.rajawali3d.control;
+
+import android.support.annotation.IntRange;
+import android.support.annotation.NonNull;
+import c.org.rajawali3d.annotations.RequiresRenderTask;
+import c.org.rajawali3d.scene.Scene;
+import c.org.rajawali3d.sceneview.SceneView;
+import c.org.rajawali3d.sceneview.Viewport;
+import c.org.rajawali3d.surface.SurfaceSize;
+import c.org.rajawali3d.surface.SurfaceView;
+
+/**
+ * API for external clients to manage the interaction between the {@link SurfaceView} rendering context and all
+ * {@link Scene}s and {@link SceneView}s associated with it. This interface is available only when a viable
+ * render environment (thread, surface, context, and api) is in place.
+ *
+ * Methods are provided for setting the frame rate, adding/removing Scenes and SceneViews to/from lists for
+ * delegated frame operation, and for depth ordering of SceneView on-screen {@link Viewport}s.
+ *
+ * <p>
+ * In each frame, {@link SceneView}s are rendered in list order resulting in a simple back-to-front painter's
+ * algorithm for their on-screen {@link Viewport}s (index 0 is back-most). There is no culling; even if a viewport is
+ * completely obscured behind another, it is still rendered as if fully visible.
+ * </p>
+ *
+ * @author Jared Woolston (Jared.Woolston@gmail.com)
+ */
+public interface RenderControl {
+
+    //
+    // Render context and surface
+    //
+
+    /**
+     * Gets the current RenderContext
+     *
+     * @return {@link RenderContext} - the current RenderContext enum instance
+     */
+    @NonNull
+    RenderContext getCurrentRenderContext();
+
+    /**
+     * Gets the current overall render surface size in pixels.
+     *
+     * @return {@link SurfaceSize} instance containing current size dimensions
+     */
+    @NonNull
+    SurfaceSize getSurfaceSize();
+
+    //
+    // Concurrency and synchronization
+    //
+
+    /**
+     * Checks whether the calling thread is the render thread.
+     *
+     * @return {@code true} if the calling thread is the render thread.
+     */
+    boolean isRenderThread();
+
+    /**
+     *
+     * @param runnable
+     */
+    void queueToMainThread(Runnable runnable);
+
+    /**
+     *
+     * @param runnable
+     */
+    void queueToRenderThread(Runnable runnable);
```

Comment:
`runnable` should be annotated as `@NonNull`

Reply:
Yep will do